build-logic/src/main/groovy/org/apache/groovy/gradle/JarJarTask.groovy

```diff
@@ -169,16 +168,9 @@ class JarJarTask extends DefaultTask {
                 it.into(outputFile.parentFile)
                 it.rename { outputFile.name }
             }
-            project.ant.jar(destfile: outputFile, update: true, index: true, manifest: manifestFile, modificationtime: tstamp) {
-                manifest {
-                    // because we don't want to use JDK 1.8.0_91, we don't care and it will
-                    // introduce cache misses
-                    attribute(name: 'Created-By', value: 'Gradle')
```

Comment:
-1 on deleting this
MANIFEST.MF now contains something like:

> Created-By: 17.0.13+11-LTS (Azul Systems, Inc.)


Instead of:

> Created-By: Gradle

Which, as well as impacting Gradle cache, breaks our reproducible build goal. "Created-By" is already in the `excludedFromManifest` list, so I guess can't really be excluded, hence its forced value.

Reply:
This should be all set now.  I can promote it from draft or you can do some investigation.  Just let me know.

Reply:
+1 for all refactoring changes and removing Ant-Version, Created-By, Extension-Name, Tool. My preference is to keep OSGi changes in a separate PR, ideally with some testing, but at least to make it easier/cleaner for manual testing.